xmtp_api_http/src/http_client.rs

```diff
@@ -0,0 +1,59 @@
+use crate::{HttpClientError, XmtpHttpApiClient};
+use bytes::Bytes;
+use std::pin::Pin;
+use xmtp_proto::traits::{ApiError, Client};
+
+impl From<HttpClientError> for ApiError<HttpClientError> {
+    fn from(value: HttpClientError) -> Self {
+        ApiError::Client { source: value }
+    }
+}
+
+#[cfg_attr(not(target_arch = "wasm32"), async_trait::async_trait)]
+#[cfg_attr(target_arch = "wasm32", async_trait::async_trait(?Send))]
+impl Client for XmtpHttpApiClient {
+    type Error = HttpClientError;
+    type Stream = Pin<Box<dyn futures::Stream<Item = Result<Bytes, HttpClientError>> + Send>>;
+    async fn request(
+        &self,
+        request: http::request::Builder,
+        body: Vec<u8>,
+    ) -> Result<http::Response<Bytes>, ApiError<Self::Error>> {
+        let request = request.body(body.clone())?;
+        let (parts, _) = request.into_parts();
+
+        let req = self
+            .http_client
+            .request(parts.method.clone(), parts.uri.to_string());
```

Comment:
We need to build the URI from the [host_url](https://github.com/xmtp/libxmtp/blob/6219b8aaa3fcc05c4e4753e282a46f7522495299/xmtp_api_http/src/lib.rs#L49) defined on the `Client`. The full URI should be something like `host_url + parts.uri.to_string()`. The request builder only contains the endpoint path w/o the base path so far

There are probably also combinators on the [Uri](https://docs.rs/http/latest/http/uri/struct.Uri.html) type that could help you out here, and you could directly set the `host`